contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,309 @@
+    event BribeCreated(
+        address gauge,
+        address rewardToken,
+        uint256 maxRewardPerVote,
+        uint256 totalRewardAmount
+    );
+    event TotalRewardAmountUpdated(uint256 extraTotalRewardAmount);
+    event NumberOfPeriodsUpdated(uint8 extraNumberOfPeriods);
+    event RewardPerVoteUpdated(uint256 newMaxRewardPerVote);
+    event RescueTokens(address token, uint256 amount, address receiver);
+
+    ////////////////////////////////////////////////////
+    /// --- CONSTRUCTOR && INITIALIZATION
+    ////////////////////////////////////////////////////
+    constructor(
+        uint256 _targetChainId,
+        address _campaignRemoteManager,
+        address _rewardToken,
+        address _gauge
+    ) {
+        targetChainId = _targetChainId;
+        campaignRemoteManager = _campaignRemoteManager;
+        rewardToken = _rewardToken;
+        gauge = _gauge;
+
+        // Prevent implementation contract to be governed
+        _setGovernor(address(0));
+    }
+
+    function initialize(
+        address _strategist,
+        uint16 _fee,
+        address _feeCollector
+    ) external initializer {
+        _setStrategistAddr(_strategist);
+        _setFee(_fee);
+        _setFeeCollector(_feeCollector);
+    }
+
+    ////////////////////////////////////////////////////
+    /// --- MUTATIVE FUNCTIONS
+    ////////////////////////////////////////////////////
+    /// @notice Create a new campaign on VotemarketV2
+    /// @dev This will use all token available in this contract
+    /// @param numberOfPeriods Duration of the campaign
+    /// @param maxRewardPerVote Maximum reward per vote
+    /// @param blacklist  List of addresses to exclude from the campaign
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function createCampaign(
+        uint8 numberOfPeriods,
+        uint256 maxRewardPerVote,
+        address[] calldata blacklist,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId == 0, "Campaign already created");
+        require(numberOfPeriods > 1, "Invalid number of periods");
+        require(maxRewardPerVote > 0, "Invalid reward per vote");
+
+        // Cache current rewardToken balance
+        uint256 balance = IERC20(rewardToken).balanceOf(address(this));
+        require(balance > 0, "No reward to manage");
+
+        // Handle fee (if any)
+        balance = _handleFee(balance);
+
+        // Approve the balance to the campaign manager
+        IERC20(rewardToken).approve(campaignRemoteManager, balance);
+
+        // Create a new campaign
+        ICampaingRemoteManager(campaignRemoteManager).createCampaign{
+            value: bridgeFee
+        }(
+            ICampaingRemoteManager.CampaignCreationParams({
+                chainId: targetChainId,
+                gauge: gauge,
+                manager: address(this),
```

Comment:
Is this the manager address on mainnet or on L2? I tried looking at the docs and couldn't find anything relevant

Reply:
The last time I discussed with the StakeDAO team was on the mainnet and L2. Let me ask again.

Reply:
Confirmed by StakeDAO team, both mainnet and L2 👍

Reply:
From what I understand, no. The only reason we might need it could be to close a campaign, so 24 weeks after the campaign ends. However, as this implies token bridging and so on, that is something that's not available on the `ICampaignRemoteManager`. 
As it should not happen really often, we can ask the StakeDAO team to close it for us.